src/core/config/index.js

```diff
@@ -0,0 +1,144 @@
+  showExtensions: false,
+  showCommonExtensions: false,
+  withCredentials: undefined,
+  requestSnippetsEnabled: false,
+  requestSnippets: {
+    generators: {
+      // eslint-disable-next-line camelcase
+      curl_bash: {
+        title: "cURL (bash)",
+        syntax: "bash",
+      },
+      // eslint-disable-next-line camelcase
+      curl_powershell: {
+        title: "cURL (PowerShell)",
+        syntax: "powershell",
+      },
+      // eslint-disable-next-line camelcase
+      curl_cmd: {
+        title: "cURL (CMD)",
+        syntax: "bash",
+      },
+    },
+    defaultExpanded: true,
+    languages: null, // e.g. only show curl bash = ["curl_bash"]
+  },
+  supportedSubmitMethods: [
+    "get",
+    "put",
+    "post",
+    "delete",
+    "options",
+    "head",
+    "patch",
+    "trace",
+  ],
+  queryConfigEnabled: false,
+
+  // Initial set of plugins ( TODO rename this, or refactor - we don't need presets _and_ plugins. Its just there for performance.
+  // Instead, we can compile the first plugin ( it can be a collection of plugins ), then batch the rest.
+  presets: [ApisPreset],
+
+  // Plugins; ( loaded after presets )
+  plugins: [],
+
+  pluginsOptions: {
+    // Behavior during plugin registration. Can be :
+    // - legacy (default) : the current behavior for backward compatibility – last plugin takes precedence over the others
+    // - chain : chain wrapComponents when targeting the same core component
+    pluginLoadType: "legacy",
+  },
+
+  // Initial state
+  initialState: {},
+
+  // Inline Plugin
+  fn: {},
+  components: {},
+
+  syntaxHighlight: {
+    activated: true,
+    theme: "agate",
+  },
+}
+
+export const getConfigs = (opts) => {
+  const queryConfig = opts.queryConfigEnabled ? parseSearch() : {}
+
+  const combinedConfig = deepExtend({}, defaultConfig, opts, queryConfig)
+
+  const storeConfig = {
+    system: {
+      configs: combinedConfig.configs,
+    },
+    plugins: combinedConfig.presets,
+    pluginsOptions: combinedConfig.pluginsOptions,
+    state: deepExtend(
+      {
+        layout: {
+          layout: combinedConfig.layout,
+          filter: combinedConfig.filter,
+        },
+        spec: {
+          spec: "",
+          // support Relative References
+          url: combinedConfig.url,
+        },
+        requestSnippets: combinedConfig.requestSnippets,
+      },
+      combinedConfig.initialState
+    ),
+  }
+
+  if (combinedConfig.initialState) {
+    // if the user sets a key as `undefined`, that signals to us that we
+    // should delete the key entirely.
+    // known usage: Swagger-Editor validate plugin tests
+    for (var key in combinedConfig.initialState) {
+      if (
+        Object.prototype.hasOwnProperty.call(
+          combinedConfig.initialState,
+          key
+        ) &&
+        combinedConfig.initialState[key] === undefined
+      ) {
+        delete storeConfig.state[key]
+      }
+    }
+  }
+
+  return { queryConfig, combinedConfig, storeConfig }
```

Comment:
We need to return all of these, as 
1. we get some options from `combinedConfig`
2. `storeConfig` is needed to initialise `System/Store` 
3. `queryConfig` is later needed for another configuration merge because it takes precedence over other configs.

Reply:
Addressed by dissecting and composing the config again together in https://github.com/swagger-api/swagger-ui/pull/9815